bin/p2-label/main.go

```diff
@@ -0,0 +1,169 @@
+
+	cmdShow       = kingpin.Command(CmdShow, "Show labels that apply to a particular entity (type, ID)")
+	showLabelType = cmdShow.Flag("labelType", "The type of label to adjust. Sometimes called the \"label tree\".\n Supported types can be found here: https://godoc.org/github.com/square/p2/pkg/labels#pkg-constants").Short('t').Required().String()
+	showID        = cmdShow.Flag("id", "The ID of the entity to show labels for.").Short('i').Required().String()
+
+	// autoConfirm captures the confirmation desire abstractly across commands
+	autoConfirm = false
+)
+
+const (
+	CmdApply = "apply"
+	CmdShow  = "show"
+)
+
+func main() {
+	cmd, opts := flags.ParseWithConsulOptions()
+	client := kp.NewConsulClient(opts)
+	applicator := labels.NewConsulApplicator(client, 3)
+	exitCode := 0
+
+	switch cmd {
+	case CmdShow:
+		labelType, err := labels.AsType(*showLabelType)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error while parsing label type. Check the commandline. \n%v\n", err)
+			break
+		}
+
+		labelsForEntity, err := applicator.GetLabels(labelType, *showID)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Got error while querying labels. %v\n", err)
+			break
+		}
+		fmt.Printf("%s/%s: %s\n", labelType, *showID, labelsForEntity.Labels.String())
+		return
+	case CmdApply:
+		autoConfirm = *applyAutoConfirm
+
+		labelType, err := labels.AsType(*applyLabelType)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Unrecognized type %s. Check the commandline and documentation.\nhttps://godoc.org/github.com/square/p2/pkg/labels#pkg-constants\n", *applyLabelType)
+			break
+		}
+
+		subject, err := klabels.Parse(*applySubjectSelector)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error while parsing subject label. Check the syntax.\n%v\n", err)
+			break
+		}
+
+		additive := klabels.Set(*applyAddititiveLabels)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error while parsing additive label set. Check the syntax.\n%v\n", err)
+			break
+		}
+		destructive := klabels.Set(*applyDestructiveLabels)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error while parsing destructive label set. Check the syntax.\n%v\n", err)
+			break
+		}
+
+		cachedMatch := false
+		matches, err := applicator.GetMatches(subject, labelType, cachedMatch)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error while finding label matches. Check the syntax.\n%v\n", err)
+			break
+		}
+
+		additiveSelector := additive.AsSelector()
+		destructiveSelector := destructive.AsSelector()
+
+		if !additiveSelector.Empty() {
+			fmt.Printf("labels to be added: %s\n", additive.String())
+		}
+
+		if !destructiveSelector.Empty() {
+			fmt.Printf("labels to be removed: %s\n", destructive.String())
+		}
+
+		var labelsForEntity labels.Labeled
+		for _, match := range matches {
+			entityID := match.ID
+
+			err := applyLabels(applicator, entityID, labelType, additiveSelector, destructiveSelector)
+			if err != nil {
+				fmt.Printf("Encountered err during labeling, %v", err)
+				exitCode = 1
```

Comment:
intentionally no `continue` here. The operator likely wants to see the state of the node as it is labelled after an error condition.

Also note that we continue through this loop even when we see errors on individual entities.